proto/cluster.proto

```diff
@@ -41,7 +41,7 @@ service ClusterService {
   // Finds all Clusters. Supports pagination, and sorting on certain fields.
   rpc ListCluster(ListClustersRequest) returns (ListClustersResponse) {
     option (google.api.http) = {
-      get: "/apis/v1alpha1/clusters"
+      get: "/apis/v1alpha1/clusters/{namespace}"
```

Comment:
@Jeffwan it makes more sense to include `namespace` in the url path. Maybe we should do the same for get and delete requests as well.

Reply:
Do you prefer to make it part of the request path or parameter? 
1.  if that's part of url, that's kind of binding to Kubernetes. (Even we know it uses Kubernetes underneath.)
2. Should we make other configuration like runtimes consistent with cluster setting?

Reply:
>if that's part of url, that's kind of binding to Kubernetes. (Even we know it uses Kubernetes underneath.)

I agree, either way we will have `namespace` as a parameter. This is for only HTTP request url path. 

Plus, if `namespace` is not in the request url path, HTTP server will get confused between get and list. It will think it's a get but with an empty cluster name

>Should we make other configuration like runtimes consistent with cluster setting?

You are referring to compute templates right? Yes, we need that as well. I can pick it up next.

Reply:
Created [an issue](https://github.com/ray-project/kuberay/issues/238) to support namespace scope for compute templates

Reply:
The default namespace is still supported because for the underlying GRPC server, namespace is optional in the request (all fields in proto3 are optional). HTTP request URL is only used to [construct a GRPC request](https://github.com/ray-project/kuberay/blob/master/proto/go_client/cluster.pb.gw.go#L336). E.g. 
`http://127.0.0.1:8888/apis/v1alpha1/clusters/` returns all ray clusters under the default namespace
`http://127.0.0.1:8888/apis/v1alpha1/clusters/foo` returns all ray clusters under the namespace `foo`

If we config namespace as the query parameter,  I noticed that GRPC misinterprets `http://127.0.0.1:8888/apis/v1alpha1/clusters/?namespace=foo` as `http://127.0.0.1:8888/apis/v1alpha1/clusters/?namespace=foo&name=""`, so it will construct a `GetClusterRequest` with an empty cluster name. That will result in an error when a user try to list the clusters under a specific namespace. I'm not sure how I can get around with this.

Logically, I think having the namespace as a path parameter is probably better than the query parameter here since the Ray cluster resource object is tied to Kubernetes namepsace. Having a default namespace is a bit odd, but this is just for the backward compatiblity. 

Please let me know what you think.

Reply:
`http://127.0.0.1:8888/apis/v1alpha1/clusters/foo` this is a little bit confusing? If we like to include namespace in the path, should url path like `http://127.0.0.1:8888/apis/v1alpha1/namespace/${namespace}/clusters/${clustername}`? 

> Having a default namespace is a bit odd, but this is just for the backward compatibility.

I agree on this part. Since this is not being used by lots of corps, we can make necessary breaking change and solve the problem by bumping versions like `v1alpha2` etc.

Reply:
This is being used.. Either id or namespace + name can be used to fetch the cluster

Reply:
As we discussed offline, this `id` is no longer used. `namesapce` + `name` will uniquely identify a Ray cluster